filebeat/input/netmetrics/tcp.go

```diff
@@ -0,0 +1,224 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package netmetrics
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"runtime"
+	"strconv"
+	"time"
+
+	"github.com/rcrowley/go-metrics"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/monitoring"
+	"github.com/elastic/elastic-agent-libs/monitoring/adapter"
+)
+
+// TCP handles the TCP metric reporting.
+type TCP struct {
+	done chan struct{}
+
+	lastPacket time.Time
+
+	device         *monitoring.String // name of the device being monitored
+	packets        *monitoring.Uint   // number of packets processed
+	bytes          *monitoring.Uint   // number of bytes processed
+	rxQueue        *monitoring.Uint   // value of the rx_queue field from /proc/net/tcp{,6} (only on linux systems)
+	arrivalPeriod  metrics.Sample     // histogram of the elapsed time between packet arrivals
+	processingTime metrics.Sample     // histogram of the elapsed time between packet receipt and publication
+}
+
+// NewTCPMetrics returns an input metric for the TCP processor.
+func NewTCPMetrics(reg *monitoring.Registry, device string, poll time.Duration, log *logp.Logger) *TCP {
```

Comment:
```suggestion
// NewTCP returns an input metric for the TCP processor.
func NewTCP(reg *monitoring.Registry, device string, poll time.Duration, log *logp.Logger) *TCP {
```

Reply:
still in my understanding the monitoring.Registry as it needs to bear the appropriate input name and id is bound to the collector and specifically it's lifetime boundaries is during the running state of the former; but since you were the original author of the tcp and udp metrics I trust your experience and I will proceed with your proposal

Reply:
as said above, I just pushed a commit with your proposal @efd6 . Please feel free to either resolve this conversation or follow up with more proposals 🙂